### ISSUE TITLE
kubedns not-ready causes test: [k8s.io] SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run [Conformance] failed all the time in ci-kubernetes-e2e-gce-serial

### ISSUE THREAD

**Issue by dchen1107** [kubernetes/kubernetes#45991]
https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-serial is very flaky since 2017-05-12 00:47 PDT took 5h5m due to the test:

[k8s.io] SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run [Conformance] 6m11s

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduling/predicates.go:109
Expected error:
    <*errors.errorString | 0xc42001c270>: {
        s: "1 / 30 pods in namespace \"kube-system\" are NOT in RUNNING and READY state in 5m0s\nPOD                       NODE                            PHASE   GRACE CONDITIONS\nkube-dns-2473071502-q95wd bootstrap-e2e-minion-group-hphc Running       [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2017-05-12 01:56:14 -0700 PDT  } {Ready False 0001-01-01 00:00:00 +0000 UTC 2017-05-12 01:56:19 -0700 PDT ContainersNotReady containers with unready status: [kubedns]} {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2017-05-12 01:56:14 -0700 PDT  }]\n",
    }
    1 / 30 pods in namespace "kube-system" are NOT in RUNNING and READY state in 5m0s
    POD                       NODE                            PHASE   GRACE CONDITIONS
    kube-dns-2473071502-q95wd bootstrap-e2e-minion-group-hphc Running       [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2017-05-12 01:56:14 -0700 PDT  } {Ready False 0001-01-01 00:00:00 +0000 UTC 2017-05-12 01:56:19 -0700 PDT ContainersNotReady containers with unready status: [kubedns]} {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2017-05-12 01:56:14 -0700 PDT  }]
    
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduling/predicates.go:99
```
One example of failed builds is: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-serial/1059/

The flakiness currently blocks our 1.7 alpha-4 build.
cc/ @kubernetes/kubernetes-release-managers @kubernetes/sig-network-bugs @bowei

**Comment by MrHohn** [comment]
To clarify, it is not "kubedns not-ready" that causes failure all the time.

For instance, this error didn't appear in the latest runs of [ci-kubernetes-e2e-gce-serial](https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-serial) (from [1076](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-serial/1076/) to [1084](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-serial/1084/)).

**Comment by MrHohn** [comment]
Found something interesting in test: ["Restart [Disruptive] should restart all nodes and ensure all nodes and pods recover"](https://github.com/kubernetes/kubernetes/blob/61b7b3fb660d09917af0b843c1b151279ea574ef/test/e2e/restart.go#L76)

Below is the log extracted from https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-serial/1060/?log#log:
```
...
I0512 09:48:14.936] STEP: ensuring the same number of nodes exist after the restart
I0512 09:48:14.936] STEP: ensuring the same number of pods are running and ready after restart
I0512 09:48:16.936] May 12 09:48:16.936: INFO: Waiting up to 4m57.999726439s for 0 pods to be running and ready, or succeeded: []
I0512 09:48:16.936] May 12 09:48:16.936: INFO: Wanted all 0 pods to be running and ready, or succeeded. Result: true. Pods: []
I0512 09:48:16.936] [AfterEach] [k8s.io] Restart [Disruptive]
I0512 09:48:16.937]   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:130
I0512 09:48:16.937] May 12 09:48:16.936: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
I0512 09:48:16.939] STEP: Destroying namespace "e2e-tests-restart-p0l8k" for this suite
...
```
After restart, this test should wait up to 5 minutes for all system pods running and ready again. But the logs indicate it wait for nothing. Although the following test will again wait up to 1 minute for all pods running, it is usually not enough. Hence it becomes very flaky.

Checked the code and found there is a mistake in https://github.com/kubernetes/kubernetes/blob/61b7b3fb660d09917af0b843c1b151279ea574ef/test/e2e/restart.go#L136, which always returns empty podNames.

Will send a PR to fix this case and hopefully reduce the flakiness.

In short don't think this is a blocker.

**Comment by MrHohn** [comment]
Didn't see this specific failure appear on [ci-kubernetes-e2e-gce-serial](https://k8s-testgrid.appspot.com/google-gce#gce-serial&width=5) after #46011 was merged. It also seems that failures have been narrowed down because we avoid those random failures.

Closing this as other failures in this test should be tracked by other issues.

/assign
